zsh-system-clipboard.zsh

```diff
@@ -40,12 +40,15 @@ if [[ "$ZSH_SYSTEM_CLIPBOARD_METHOD" == "" ]]; then
 			fi
 			;;
 		linux*|freebsd*)
-			if [[ "$DISPLAY" == "" ]]; then
+			if _zsh_system_clipboard_command_exists wl-copy && ( [[ "$WAYLAND_DISPLAY" != "" ]] || [[ "$XDG_SESSION_TYPE" == wayland ]] ); then
 				ZSH_SYSTEM_CLIPBOARD_METHOD="wlc"
-			elif _zsh_system_clipboard_command_exists xclip; then
+			elif _zsh_system_clipboard_command_exists xclip && [[ "$DISPLAY" != "" ]]; then
 				ZSH_SYSTEM_CLIPBOARD_METHOD="xcc"
-			elif _zsh_system_clipboard_command_exists xsel; then
+			elif _zsh_system_clipboard_command_exists xsel && [[ "$DISPLAY" != "" ]]; then
 				ZSH_SYSTEM_CLIPBOARD_METHOD="xsc"
+			elif [[ "$SSH_TTY" != "" ]] || [[ "$XDG_SESSION_TYPE" == tty ]]; then
```

Comment:
I use often `ssh -Y` which makes `"$XDG_SESSION_TYPE" == tty`, `$SSH_TTY != ""` and `$DISPLAY != ""` true. This enables me to use zsh-system-clipboard with `xsel` installed on the server - meaning the line checking or `xsel` and a non empty DISPLAY will succeed for me. I still don't like this line however - users using `ssh -Y` without `xsel` or `xclip` installed on the target host would fall into this conditional and it'll be hard for them to debug why it zsh-system-clipboard doesn't work because you don't suggest them to install xclip etc.

You could also check in this line if also `$DISPLAY` is empty, but I still don't like this silent `return`.. If you don't want to use zsh-system-clipboard on a remote host, don't install it there at all.

Reply:
> If you don't want to use zsh-system-clipboard on a remote host, don't install it there at all.

Fair